htdocs/class/xoopshttpget.php

```
@@ -50,52 +50,177 @@ public function __construct($url)
+        if (filter_var($host, FILTER_VALIDATE_IP)) {
+            $ips = [$host];
         } else {
-            $httpcode = curl_getinfo($curlHandle, CURLINFO_HTTP_CODE);
-            if (200 != $httpcode) {
+            // @-suppressed: an unresolvable host emits a benign "host not found"
+            // warning whose false return we handle immediately (fail closed, R-030).
+            $ips = @gethostbynamel($host);
+            if ($ips === false || $ips === []) {
+                return null;
+            }
+        }
```

PHPMD (2.15.0)

[error] The method resolveAllowed uses an else expression. Else clauses are basically not necessary and you can simplify the code by not using them. (undefined)

(ElseExpression)